browser/resources/settings/getting_started_page/brave_account_create_dialog.ts

```diff
@@ -0,0 +1,192 @@
+      }
+
+      :host([category=Weak]) {
+        --primary-color: var(--leo-color-systemfeedback-error-icon);
+        --secondary-color: var(--leo-color-systemfeedback-error-background);
+      }
+
+      :host([category=Medium]) {
+        --primary-color: var(--leo-color-systemfeedback-warning-icon);
+        --secondary-color: var(--leo-color-systemfeedback-warning-background);
+      }
+
+      :host([category=Strong]) {
+        --primary-color: var(--leo-color-systemfeedback-success-icon);
+        --secondary-color: var(--leo-color-systemfeedback-success-background);
+      }
+
+      .bar {
+        background-color: var(--secondary-color);
+        border-radius: var(--leo-radius-m);
+        height: 4px;
+        transition: 750ms;
+        width: 376px;
+      }
+
+      .strength {
+        background-color: var(--primary-color);
+        border-radius: var(--leo-radius-m);
+        height: 100%;
+        transition: 750ms;
+        width: calc(1% * var(--strength));
+      }
+
+      .text {
+        color: var(--primary-color);
+        font: var(--leo-font-small-regular);
+        transition: 750ms;
+      }
+    `
+  }
+
+  override render() {
+    return html`
+      <div class="bar">
+        <div class="strength" style="--strength: ${this.strength}"></div>
+      </div>
+      <div class="text">
+        ${this.i18n(`braveAccountPasswordStrengthMeter${this.category}`)}
+      </div>
+    `
+  }
+
+  static override get properties() {
+    return {
+      category: { type: String, reflect: true },
+      strength: { type: Number },
+    }
+  }
+
+  override updated(changedProperties: Map<PropertyKey, unknown>) {
+    if (changedProperties.has('strength')) {
+      this.category = this.strength < 60
+        ? 'Weak'
+        : this.strength < 100
+          ? 'Medium'
+          : 'Strong'
+    }
+  }
+
+  protected category: 'Weak' | 'Medium' | 'Strong' = 'Weak'
+  protected strength: number = 0
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'password-strength-meter': PasswordStrengthMeter
+  }
+}
+
+customElements.define(
+  PasswordStrengthMeter.is, PasswordStrengthMeter)
+
+export class SettingsBraveAccountCreateDialogElement extends CrLitElement {
+  static get is() {
+    return 'settings-brave-account-create-dialog'
+  }
+
+  static override get styles() {
+    return getCss()
+  }
+
+  override render() {
+    return getHtml.bind(this)()
+  }
+
+  static override get properties() {
+    return {
+      email: { type: String },
+      isCheckboxChecked: { type: Boolean },
+      isEmailBraveAlias: { type: Boolean },
+      isEmailValid: { type: Boolean },
+      password: { type: String },
+      passwordConfirmation: { type: String },
+      passwordStrength: { type: Number },
+    }
+  }
+
+  protected onEmailInput(detail: { value: string }) {
+    this.email = detail.value
+    this.isEmailValid = isEmailValid(this.email)
+    this.isEmailBraveAlias = this.email.endsWith('@bravealias.com')
```

Comment:
We should do a case-insensitive comparison for this one so that `@BRAVEALIAS.com` is also rejected.

Reply:
```js
this.isEmailBraveAlias = (/@bravealias.com$/i).test(this.email)
```